modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy

```diff
@@ -573,8 +573,16 @@ class PodSpecBuilder {
         // -- host path volumes
         for( PodHostMount entry : hostMounts ) {
             final name = nextVolName()
-            mounts << [name: name, mountPath: entry.mountPath]
-            volumes << [name: name, hostPath: [path: entry.hostPath]]
+            if (entry.readOnly) {
+                mounts << [name: name, mountPath: entry.mountPath, readOnly: true]
+            } else {
+                mounts << [name: name, mountPath: entry.mountPath]
+            }
```

Comment:
Consider something like to improve readability  

```
if( entry.readOnly ) 
  mounts.readOnly = true
```

Reply:
this fails because of the compilestatic annotation. If you think the code is not very clear I can add a comment but IMO that is completely redundant because the if is very clear - plus the branch is short

Reply:
then you can use 

```
if( entry.readOnly ) 
  mounts.put('readOnly', true)
```

Reply:
IMO this is less obvious than this `mounts << [name: name, mountPath: entry.mountPath, readOnly: true]` but I will go with it if you prefer it

Reply:
Sounds good

Reply:
@pditommaso done